internal/ethapi/api.go

```diff
@@ -1202,6 +1202,11 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 	if err := args.CallDefaults(gasCap, header.BaseFee, b.ChainConfig().ChainID); err != nil {
 		return 0, err
 	}
+	// when gas is over header.GasLimit, just recap with header.GasLimit
+	if uint64(*args.Gas) > header.GasLimit {
+		*args.Gas = (hexutil.Uint64)(header.GasLimit)
```

Comment:
Please change it so that `args.Gas` is cleared to `nil` if it was previously `nil` before the call to `args.CallDefaults`. @karalabe and I think it's cleaner this way.

Reply:
The reason is that gasestimator.Estimate is meant to be self-contained (and already has support for capping to the block gas limit). This PR as is currently leaks one of the decisions out to the outside world and that's going to get messy long term, so our best solution is to "undo" the bad default that "CallDefaults" sets. That way at least the "error" is self contained to this few lines of code and doesn't keep bubbling further on the stack.